httpclient5/src/main/java/org/apache/hc/client5/http/entity/compress/ContentCodecRegistry.java

```diff
@@ -0,0 +1,171 @@
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+
+import org.apache.hc.client5.http.entity.DeflateInputStream;
+import org.apache.hc.core5.annotation.Contract;
+import org.apache.hc.core5.annotation.Internal;
+import org.apache.hc.core5.annotation.ThreadingBehavior;
+import org.apache.hc.core5.http.HttpEntity;
+import org.brotli.dec.BrotliInputStream;
+
+
+/**
+ * Run-time catalogue of built-in and Commons-Compress
+ * {@linkplain Encoder encoders} / {@linkplain Decoder decoders}.
+ *
+ * <p>Entries are wired once at class-load time and published through an
+ * unmodifiable map, so lookups are lock-free and thread-safe.</p>
+ *
+ * @since 5.6
+ */
+@Internal
+@Contract(threading = ThreadingBehavior.STATELESS)
+public final class ContentCodecRegistry {
+
+    private static final Map<ContentCoding, Codec> REGISTRY = build();
+
+    private static Map<ContentCoding, Codec> build() {
+        final Map<ContentCoding, Codec> m =
+                new EnumMap<>(ContentCoding.class);
+
+        /* 1. Built-ins ------------------------------------------------- */
+        m.put(ContentCoding.GZIP,
+                new Codec(
+                        // encoder
+                        org.apache.hc.client5.http.entity.GzipCompressingEntity::new,
+                        // decoder
+                        GZIPInputStream::new));
+        m.put(ContentCoding.DEFLATE,
+                new Codec(
+                        org.apache.hc.client5.http.entity.DeflateCompressingEntity::new,
+                        DeflateInputStream::new));
+
+        /* 2. Commons-Compress extras ---------------------------------- */
+        if (CommonsCompressSupport.isPresent()) {
+            for (final ContentCoding c : Arrays.asList(
+                    ContentCoding.BROTLI,
+                    ContentCoding.ZSTD,
+                    ContentCoding.XZ,
+                    ContentCoding.LZMA,
+                    ContentCoding.LZ4_FRAMED,
+                    ContentCoding.LZ4_BLOCK,
+                    ContentCoding.BZIP2,
+                    ContentCoding.PACK200,
+                    ContentCoding.DEFLATE64)) {
+
+                if (CommonsCompressDecoderFactory.runtimeAvailable(c.token())) {
+                    m.put(c, new Codec(e -> new CommonsCompressingEntity(e, c.token()), new CommonsCompressDecoderFactory(c.token())
+                    ));
+                }
+            }
+        }
+
+        /* 3. Native Brotli fallback (decode-only) ---------------------- */
+        if (!m.containsKey(ContentCoding.BROTLI)
+                && CommonsCompressDecoderFactory.runtimeAvailable(ContentCoding.BROTLI.token())) {
+            m.put(ContentCoding.BROTLI,
+                    Codec.decodeOnly(BrotliInputStream::new));
+        }
+
+        return Collections.unmodifiableMap(m);
+    }
+
+
+    /**
+     * Returns the {@link Codec} (or {@code null}).
+     */
+    public static Codec codec(final ContentCoding coding) {
```

Comment:
@arturobernalg Here we have a scope violation: a public method exposing a package private class. Either make the method package private or make the class public. It is internal anyway.